<launch>
  <arg name="direct_model" default="false" />
  <arg name="direct_model_name" />
  <arg name="type" default="octo" />
  <arg name="model" default="v1_default" />
  <arg name="headless" default="False" />
  <arg name="description_mode" default="urdf"/>
  <arg name="need_joint_state" default="true" />
  <arg name="node_suffix" default="" />

  <group ns="tiger" >
    <node name="robot_state_publisher_$(arg node_suffix)" pkg="robot_state_publisher" type="state_publisher"/>

    <rosparam file="$(find tiger)/config/$(arg type)/RvizInit.yaml" command="load"/>
    <group if="$(arg need_joint_state)">
      <param name="use_gui" value="true" />
      <node name="joint_state_publisher_$(arg node_suffix)" pkg="joint_state_publisher" type="joint_state_publisher" >
        <param name="robot_description"
               command="$(find xacro)/xacro '$(find tiger)/robots/$(arg type)/$(arg model)/robot.urdf.xacro'"/>
      </node>
    </group>

    <node unless="$(arg need_joint_state)" name="aerial_robot_model_$(arg node_suffix)" pkg="aerial_robot_model" type="rotor_tf_publisher"/>

  </group>

  <include file="$(find aerial_robot_model)/launch/aerial_robot_model.launch" >
    <arg name="headless" value="$(arg headless)" />
    <arg unless="$(arg direct_model)" name="robot_model" value="$(find tiger)/robots/$(arg type)/$(arg model)/robot.$(arg description_mode).xacro"/>
    <arg if="$(arg direct_model)" name="robot_model" value="$(arg direct_model_name).$(arg description_mode).xacro" />
    <arg name="node_suffix" value="$(arg node_suffix)" />
  </include>

</launch>
